packages/sdk/src/Fuel.ts

```diff
@@ -14,6 +15,33 @@ const FuelWeb3Privates: {
 } = {};
 
 export class Fuel extends FuelWalletConnection {
+  readonly walletProviders: Array<WalletProvider> = [];
+
+  constructor(providerConfig: WalletProvider) {
+    super(providerConfig.name);
+    this.walletProviders = [providerConfig];
+  }
+
+  addWalletProvider(walletProvider: WalletProvider): void {
+    this.walletProviders.push(walletProvider);
+  }
+
+  listWalletProviders(): Array<WalletProvider> {
+    return this.walletProviders;
+  }
+
+  selectWalletProvider(providerName: string): void {
+    const providerConfig = this.walletProviders.find(
+      (provider) => provider.name === providerName
+    );
+
+    if (!providerConfig) {
+      throw new Error(`Wallet provider ${providerName} not found`);
+    }
+
+    this.targetWallet = providerConfig.name;
```

Comment:
NIT: for me this looks like accessing a private contract of the `WindowConnection` class. In the constructor of `Fuel` you call the `WindowConnection` super constructor and pass `providerConfig.name`, but here you directly set `providerConfig.name` to a `WindowConnection`'s field. This is a bit error prone, cause someone later might use `targetWallet` in `WindowConnection`'s in a different way, then he will also need to change the behavior here. Maybe it would be better to add a `setupTargetWallet` method in the `WindowConnection` and call it in both places.

I have a lack of knowledge in TS, not sure if that's possible. Just decided to share some thoughts :)